erpnext/setup/setup_wizard/operations/taxes_setup.py

```diff
@@ -160,12 +193,14 @@ def get_or_create_account(company_name, account):
 
 	existing_accounts = frappe.get_list('Account',
 		filters={
-			'company': company_name,
-			'root_type': root_type
+			'account_name': account.get('account_name'),
+			'account_number': account.get('account_number', ''),
+			'company': company_name
 		},
 		or_filters={
-			'account_name': account.get('account_name'),
-			'account_number': account.get('account_number')
+			'company': company_name,
+			'root_type': root_type,
+			'is_group': 0
```

Comment:
Company is both in `filters` and `or_filters`. The intention for having `account_name` and `account_number` in `or_filters` was that each should be unique and one is enough to identify the account. Having both in filters, especially `'account_number' == ''` might not work.

Reply:
@deepeshgarg007 I agree with Raffael, `account_number` and `account_name` should be in `or_filters`